packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -40,60 +39,15 @@ export default async function proxyPreviewMode(
   ...args: ProxyPreviewModeHandlerArgs
 ): Promise<NextResponse<ProxyPreviewModeResponse> | void> {
   return match(args)
-    .with(routeHandlerPattern, args => proxyPreviewModeRouteHandler(...args))
+    .with(routeHandlerPattern, () => {
+      throw new Error(
+        'Cannot request proxy preview mode endpoint from an API handler registered in `app`. Move your Makeswift API handler to the `pages/api` directory',
+      )
```

Comment:
Is this error shown to the user in a useful way? Should we instead return an error HTTP response?

Reply:
Shifted towards returning HTTP errors, as well as reflecting the error in the next console.